tests/phpunit/tests/user.php

```diff
@@ -915,6 +915,69 @@ public function test_wp_insert_user_should_reject_user_login_that_matches_existi
 		$this->assertSame( 'existing_user_email_as_login', $user_id->get_error_code() );
 	}
 
+	/**
+	 * @ticket 57967
+	 */
+	public function test_wp_update_user_should_allow_user_login_with_same_user_email() {
+		$user_email = str_repeat( 'a', 5 ) . '@example.com';
+
+		$user_id = wp_insert_user(
+			array(
+				'user_login'    => $user_email,
+				'user_email'    => $user_email,
+				'user_pass'     => 'whatever',
+				'user_nicename' => 'whatever',
+			)
+		);
+
+		$existing_id = wp_update_user(
+			array(
+				'ID'         => $user_id,
+				'first_name' => 'Bob',
+			)
+		);
+
+		$this->assertNotWPError( $existing_id );
+		$this->assertSame( $existing_id, $user_id );
+	}
+
+	/**
+	 * @ticket 57967
+	 */
+	public function test_wp_update_user_should_reject_user_login_that_matches_existing_user_email() {
+		$user_email_a = str_repeat( 'a', 5 ) . '@example.com';
```

Comment:
```suggestion
		$user_email_a = 'aaaaa@example.com';
```
Let's just hardcode the email address here. 🙂